ServerCore/ModelBases/EventSpecificPageModel.cs

```diff
@@ -39,6 +39,17 @@ public override async Task OnPageHandlerExecutionAsync(PageHandlerExecutingConte
                 return;
             }
 
+            // Check permissions now that the role has been binded
+            await IsEventAdmin();
+            await IsEventAuthor();
```

Comment:
Now that I understand why these lines aren't a perf issue: If the LoggedInUser is null, then weird things will happen. Rather than using the static cache, I would just use the return value of these IsEventXxx functions, which will either use the cache or not but definitely be bools.

Reply:
Unfortunately, not possible.  For reasons beyond my understanding, you can't call async functions in if blocks, so I'll just have to add a branch for the logged out case.

Reply:
not even

`bool isAuthor = await IsEventAuthor();`

???